packages/ui/src/fields/Blocks/index.tsx

```diff
@@ -150,6 +133,38 @@ const BlocksFieldComponent: BlocksFieldClientComponent = (props) => {
     validate: memoizedValidate,
   })
 
+  const { clientBlocks, clientBlocksAfterFilter } = useMemo(() => {
```

Comment:
This is the same as the previous `clientBlocks = useMemo()`, but it also filters blocks now according to blocksFilterOptions returned from the form state.

We need both the filtered and unfiltered client blocks. The filtered ones are only used for the drawer. We still need the ability to display existing blocks that no longer match the filter and fail validation.